sdk/python/feast/data_source.py

```diff
@@ -512,6 +512,74 @@ def source_datatype_to_feast_value_type() -> Callable[[str], ValueType]:
         return type_map.redshift_to_feast_value_type
 
 
+class RequestDataSource(DataSource):
+    """
+    RequestDataSource that can be used to provide input features for on demand transforms
+
+    Args:
+        name: Name of the request data source
+        schema: Schema mapping from the input feature name to a ValueType
+    """
+
+    @staticmethod
+    def source_datatype_to_feast_value_type() -> Callable[[str], ValueType]:
+        raise NotImplementedError
```

Comment:
nit: I wonder if this method makes sense for this type of data source. We should probably refactor the `DataSource` interface to make it simpler and introduce a BatchDataSource somewhere in the type hierarchy.